packages/react-native-executorch/common/rnexecutorch/models/ocr/DetectorUtils.cpp

```diff
@@ -0,0 +1,654 @@
+#include "DetectorUtils.h"
+#include <algorithm>
+#include <limits>
+#include <rnexecutorch/models/ocr/Constants.h>
+#include <rnexecutorch/models/ocr/Types.h>
+#include <unordered_set>
+namespace rnexecutorch::ocr {
+
+std::pair<cv::Mat, cv::Mat> interleavedArrayToMats(std::span<const float> data,
+                                                   cv::Size size) {
+  cv::Mat mat1 = cv::Mat(size.height, size.width, CV_32F);
+  cv::Mat mat2 = cv::Mat(size.height, size.width, CV_32F);
+
+  for (std::size_t i = 0; i < data.size(); i++) {
+    const float value = data[i];
+    const int x = (i / 2) % size.width;
+    const int y = (i / 2) / size.width;
+
+    if (i % 2 == 0) {
+      mat1.at<float>(y, x) = value;
+    } else {
+      mat2.at<float>(y, x) = value;
+    }
+  }
+  return {mat1, mat2};
+}
+
+std::vector<DetectorBBox> getDetBoxesFromTextMap(cv::Mat &textMap,
+                                                 cv::Mat &affinityMap,
+                                                 float textThreshold,
+                                                 float linkThreshold,
+                                                 float lowTextThreshold) {
+  const int imgH = textMap.rows;
+  const int imgW = textMap.cols;
+  cv::Mat textScore;
+  cv::Mat affinityScore;
+  cv::threshold(textMap, textScore, textThreshold, 1, cv::THRESH_BINARY);
+  cv::threshold(affinityMap, affinityScore, linkThreshold, 1,
+                cv::THRESH_BINARY);
+  cv::Mat textScoreComb = textScore + affinityScore;
+  cv::threshold(textScoreComb, textScoreComb, 0, 1, cv::THRESH_BINARY);
+  cv::Mat binaryMat;
+  textScoreComb.convertTo(binaryMat, CV_8UC1);
+
+  cv::Mat labels, stats, centroids;
+  const int nLabels =
+      cv::connectedComponentsWithStats(binaryMat, labels, stats, centroids, 4);
+
+  std::vector<DetectorBBox> detectedBoxes;
+  for (int i = 1; i < nLabels; i++) {
+    const int area = stats.at<int>(i, cv::CC_STAT_AREA);
+    if (area < 10)
+      continue;
+
+    cv::Mat mask = (labels == i);
+    double maxVal;
+    cv::minMaxLoc(textMap, nullptr, &maxVal, nullptr, nullptr, mask);
+    if (maxVal < lowTextThreshold) {
+      continue;
+    }
+
+    cv::Mat segMap = cv::Mat::zeros(textMap.size(), CV_8U);
+    segMap.setTo(255, mask);
+
+    const int x = stats.at<int>(i, cv::CC_STAT_LEFT);
+    const int y = stats.at<int>(i, cv::CC_STAT_TOP);
+    const int w = stats.at<int>(i, cv::CC_STAT_WIDTH);
+    const int h = stats.at<int>(i, cv::CC_STAT_HEIGHT);
+    const int dilationRadius = (int)(sqrt((double)(area / MAX(w, h))) * 2.0);
+    const int sx = MAX(x - dilationRadius, 0);
+    const int ex = MIN(x + w + dilationRadius + 1, imgW);
+    const int sy = MAX(y - dilationRadius, 0);
+    const int ey = MIN(y + h + dilationRadius + 1, imgH);
```

Comment:
`MIN` and `MAX` seem to be some strange macros, rather change all occurrences to `std::min` and `std::max`.